src/machines/java/com/enderio/machines/client/rendering/blockentity/FluidTankBER.java

```diff
@@ -12,61 +15,136 @@
 import net.minecraft.client.renderer.blockentity.BlockEntityRenderer;
 import net.minecraft.client.renderer.blockentity.BlockEntityRendererProvider;
 import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.core.BlockPos;
 import net.minecraft.core.Direction;
 import net.minecraft.world.inventory.InventoryMenu;
 import net.minecraft.world.level.block.entity.BlockEntity;
 import net.minecraft.world.level.material.Fluid;
+import net.minecraft.world.level.material.Fluids;
 import net.minecraftforge.client.extensions.common.IClientFluidTypeExtensions;
-import net.minecraftforge.fluids.FluidStack;
 import net.minecraftforge.fluids.capability.templates.FluidTank;
 
 public class FluidTankBER implements BlockEntityRenderer<FluidTankBlockEntity> {
+    private static float ANIMATION_TICKS = 20.0f;
+    private static HashMap<BlockPos, AnimationInformation> map = new HashMap<BlockPos, AnimationInformation>();
+
     public FluidTankBER(BlockEntityRendererProvider.Context context) {
 
     }
 
     @Override
-    public void render(FluidTankBlockEntity blockEntity, float partialTick, PoseStack poseStack, MultiBufferSource bufferSource, int packedLight,
-        int packedOverlay) {
+    public void render(FluidTankBlockEntity blockEntity, float partialTick, PoseStack poseStack,
```

Comment:
Could I ask for more documentation around the animation implementation you've added?

Reply:
Yes of course, do you mean just comments and Javadocs, or actual documentation in like a wiki or something?

Reply:
Yeah just comments and javadocs please

Reply:
Fixed in 3010b205c2c2779c487f29d5f68ad5f32ac068e6